src/format.ts

```diff
@@ -278,35 +278,47 @@ export class ModelFormat<T extends Entity> extends Format<T> {
 		}
 
 		var result = "";
```

Comment:
The pattern of calling `convertTokens` to accumulate the `result` value is a bit awkward now. It would simplify the logic if `convertTokens` were a pure function that returns a result instead of relying on the side effect of appending to `result`.